### ISSUE TITLE
Blank window on macOS until it is resized.

### ISSUE THREAD

**Issue by dalboris** [vgc/vgc#182]
On macOS, when launching VGC Illustration, it opens maximised as expected, but the window is completely empty. You can still see the correct mouse cursor (cross cursor or normal cursor depending on where the mouse is), and you can even draw, but you can't see anything.

Workaround: resize the window. Then everything works as normal.

![image](https://user-images.githubusercontent.com/4809739/64927170-73c26080-d807-11e9-9d91-17abe7466cb3.png)

**Comment by dalboris** [comment]
After a long session of debugging which was driving me nuts, this problem seems related to OpenGL on macOS 10.14 specifically.

If I change the code of `MainWindow::setupWidgets_()` to:

```
void MainWindow::setupWidgets_()
{
    QOpenGLWidget* openGlWidget = new QOpenGLWidget();
    QPushButton* button = new QPushButton("Hello!");
    QWidget* centralWidget = new QWidget();
    QVBoxLayout* layout = new QVBoxLayout();
    layout->addWidget(openGlWidget);
    layout->addWidget(button);
    centralWidget->setLayout(layout);
    setCentralWidget(centralWidget);
}

```

then nothing is drawn:

![image](https://user-images.githubusercontent.com/4809739/65394317-7cc5ab80-dd8c-11e9-8f4c-78c90cebc7c9.png)

Until you manually resize the window:

![image](https://user-images.githubusercontent.com/4809739/65394320-89e29a80-dd8c-11e9-8fe9-329a64522156.png)

But if you comment out the line adding the QOpenGLWidget to the layout:

```
void MainWindow::setupWidgets_()
{
    QOpenGLWidget* openGlWidget = new QOpenGLWidget();
    QPushButton* button = new QPushButton("Hello!");
    QWidget* centralWidget = new QWidget();
    QVBoxLayout* layout = new QVBoxLayout();
    //layout->addWidget(openGlWidget);
    layout->addWidget(button);
    centralWidget->setLayout(layout);
    setCentralWidget(centralWidget);
}
```

Then the window is correctly painted right away, without a need to resize the window:

![image](https://user-images.githubusercontent.com/4809739/65394311-64ee2780-dd8c-11e9-8f5c-dfc8ff763cf2.png)

Others are facing similar problems:

https://forum.qt.io/topic/97427/osx-window-is-blank-until-forced-repaint
https://bugreports.qt.io/browse/QTBUG-72387
https://bugreports.qt.io/browse/QTBUG-71988

And a similar problem seems to affect glfw too:

https://github.com/glfw/glfw/issues/1334

For completeness, the following were suggested workarounds::
- Set "QSG_RENDER_LOOP=windows"
- Setting the target OSX version to 10.12 in CMake

My personal workaround was to first call `mainWindow.show()` before `app.exec()`, then call `mainWindow.showMaximized()` using a delayed timer, after `app.exec()`. This worked okay, apart from the fact that the app didn't first appear maximized, and the macOS animation from non-maximized to maximized was somehow distracting.

However, the bug seems actually fixed since Qt 5.12.1 (see QTBUG-71988), and I am still using Qt 5.12.0. So hopefully upgrading Qt is all I need. I'll try that and test again.

**Comment by dalboris** [comment]
Conclusion: using Qt 5.12.5 solves the problem :)
